### PR TITLE
feat(aws-s3-eventbridge): New Construct

### PR DESCRIPTION
**Pull request by mickychetta**
*Issue #494, if available:*

*Description of changes:*
-Created README for aws-s3-eventbridge

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 4471e91cb2de4d3bdb6d8b4275ba8b4b19356882
* Result: FAILED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6676e2e5-9e95-4a94-9f5c-ba2923478a8c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: 4471e91cb2de4d3bdb6d8b4275ba8b4b19356882
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b57fe59e-f1f8-4385-a91e-a41b907e63f1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: e0ec62bb5572c8ae45276f4b7094fa2eb87bd846
* Result: FAILED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a45041a0-c779-4975-95a6-e4b1938657c0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: e0ec62bb5572c8ae45276f4b7094fa2eb87bd846
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0312f4da-73f6-4dd4-aed0-6014d88f26d0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 28f229e9973e14a4fb24e8dae2e214f748048139
* Result: FAILED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1053d7bb-218a-4bc0-a54b-4ee23ea998bf%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: 28f229e9973e14a4fb24e8dae2e214f748048139
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8c242cce-d94e-40ab-816c-f608f607abae%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-s3-eventbridge/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+This AWS Solutions Construct implements an Amazon S3 bucket connected to an Amazon EventBridge.
+
+Here is a minimal deployable pattern definition in Typescript:
+
+``` typescript
+import { S3ToEventbridge, S3ToEventbridgeProps } from '@aws-solutions-constructs/aws-s3-eventbridge';
+import { EventbridgeToStepfunctions, EventbridgeToStepfunctionsProps } from '@aws-solutions-constructs/aws-s3-stepfunctions';
+import * as stepfunctions from '@aws-cdk/aws-stepfunctions';
+
+const startState = new stepfunctions.Pass(stack, 'StartState');
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-s3-eventbridge/README.md:34]: Why is there step functions content here? This construct passes S3 events to EventBridge, what happens to the event after that is out of scope.

File: source/patterns/@aws-solutions-constructs/aws-s3-eventbridge/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+const s3ToEventbridge = new S3ToEventbridge(this, 's3-eventbridge', {});
+const eventBus = s3ToEventbridge.eventBus;
+
+new EventbridgeToStepfunctions(this, 'test-s3-stepfunctions-stack', {
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-s3-eventbridge/README.md:39]: Huh?

File: source/patterns/@aws-solutions-constructs/aws-s3-eventbridge/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+* scope [`Construct`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.Construct.html)
+* id `string`
+* props [`S3ToEventbridgeProps`](#pattern-construct-props)
+
+## Pattern Construct Props
+
+| **Name**     | **Type**        | **Description** |
+|:-------------|:----------------|-----------------|
+|existingBucketObj?|[`s3.IBucket`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-s3.IBucket.html)|Existing instance of S3 Bucket object. If this is provided, then also providing bucketProps is an error. |
+|bucketProps?|[`s3.BucketProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-s3.BucketProps.html)|Optional user provided props to override the default props for the S3 Bucket.|
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-s3-eventbridge/README.md:64]: I think this construct "responds to S3 events" - while not directly, it's the same idea, we may want to control the events that come out of S3 that are passed on to EventBridge.

File: source/patterns/@aws-solutions-constructs/aws-s3-eventbridge/README.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+|eventRuleProps?|[`events.RuleProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-events.RuleProps.html)|Optional user provided eventRuleProps to override the defaults.|
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-s3-eventbridge/README.md:65]: This is our first construct with EventBridge receiving events. I don't think eventRuleProps is appropriate here, it is a filter on what events a consumer of events will respond to.

I do think we need the custom event bus props, existingEventProps and eventBusProps.

File: source/patterns/@aws-solutions-constructs/aws-s3-eventbridge/README.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+|deployCloudTrail?|`boolean`|Whether to deploy a Trail in AWS CloudTrail to log API events in Amazon S3. Defaults to `true`.|
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-s3-eventbridge/README.md:66]: This appears in a single construct, aws-s3-stepfunctions (OK, technically 2 as it appears in the associated wrapper, aws-s3-step-function :-), but it is not listed in DESIGN_GUIDELINES.md and not something we have considered nor discuss propagating to other constructs.

Shorter version: drop this.

**Review by biffgaut — CHANGES_REQUESTED** [review]

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 8877c3f4f94a02992403fec4415ac414079bc709
* Result: FAILED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7ba0735f-46ad-4bdd-a6d0-1f5a50da2ce5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: 8877c3f4f94a02992403fec4415ac414079bc709
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=13692d73-e27b-420b-8291-207cfade89bf%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: d3eb1403469e6c02866e4b24b21dd2d3101a8f68
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c34a4308-7df9-4992-b02b-2620a4aee93f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: d3eb1403469e6c02866e4b24b21dd2d3101a8f68
* Result: FAILED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a971ed43-cdd4-41ad-9a17-09b158bb22c0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by biffgaut** [comment]
This was handled in another PR using S3 Event Notifications.

**Pull request closed without merging**